src/prefect/tasks/azure/datafactory.py

```diff
@@ -0,0 +1,356 @@
+        self,
+        datafactory_name: str = None,
+        resource_group_name: str = None,
+        pipeline_name: str = None,
+        azure_credentials_secret: str = None,
+        parameters: Optional[Dict[str, Any]] = None,
+        last_updated_after: Optional[datetime] = None,
+        last_updated_before: Optional[datetime] = None,
+    ) -> Dict[Any, Any]:
+        """
+        Create an Azure datafactory pipeline run.
+
+        Args:
+            - datafactory_name (str): Name of the datafactory to create.
+            - resource_group_name (str): Name of the resource group.
+            - pipeline_name (str): Name of the pipeline.
+            - azure_credentials_secret (str, optional): Name of the Prefect Secret that stores
+                your Azure credentials; This Secret must be JSON string with the keys
+                `subscription_id`, `client_id`, `secret`, and `tenant`.
+            - parameters (dict, optional): The parameters to be used in pipeline.
+            - last_updated_after (datetime, optional): The time at or after which the run event
+                was updated; used to filter and query the pipeline run, and defaults to yesterday.
+            - last_updated_before (datetime, optional): The time at or before which the run event
+                was updated; used to filter and query the pipeline run and defaults to tomorrow.
+
+        Returns:
+            The pipeline run response.
+        """
+        if not datafactory_name:
+            raise ValueError("The datafactory_name must be specified.")
+        if not resource_group_name:
+            raise ValueError("The resource_group_name must be specified.")
+        if not pipeline_name:
+            raise ValueError("The pipeline_name must be specified.")
+
+        datafactory_client = _get_datafactory_client(azure_credentials_secret)
+        last_updated_after = last_updated_after or datetime.utcnow() - timedelta(days=1)
+        last_updated_before = last_updated_before or datetime.utcnow() + timedelta(
+            days=1
+        )
+
+        self.logger.info(
+            f"Preparing to run the {pipeline_name} pipeline in the "
+            f"{datafactory_name} factory under {resource_group_name}."
+        )
+        # Create a pipeline run
+        create_run = datafactory_client.pipelines.create_run(
+            resource_group_name,
+            datafactory_name,
+            pipeline_name,
+            parameters=parameters or {},
+        )
+
+        # Monitor the pipeline run
+        pipeline_run = datafactory_client.pipeline_runs.get(
+            resource_group_name, datafactory_name, create_run.run_id
+        )
+        while pipeline_run.status != "Succeeded":
+            pipeline_run = datafactory_client.pipeline_runs.get(
+                resource_group_name, datafactory_name, create_run.run_id
+            )
+            self.logger.info(f"The pipeline run status: {pipeline_run.status}")
+            time.sleep(1)
```

Comment:
I think we could have a slightly higher poll interval to reduce the possibility of rate limiting. It'd also be nice if the poll interval was configurable.